client/components/domains/register-domain-step/index.jsx

```diff
@@ -479,6 +546,12 @@ class RegisterDomainStep extends React.Component {
 		);
 	};
 
+	getAvailableTlds = () => {
+		getAvailableTlds().then( availableTlds => {
```

Comment:
Sorry, I missed this in my review, @jsnmoon, but stumbled upon an error in this code while testing `D12192-code`. If new purchases are disabled, this endpoint returns an error that is not handled here and thus propagated upstream where it returns in an error :(

Reply:
@klimeryk: Would adding a `.catch( ... )` here be sufficient? I'm not familiar with how the interface handles disabled purchases.

Reply:
Possibly. Check testing instructions in `D12192-code` to learn how to trigger this error. The example given there is enough to have new purchases disabled.

Reply:
The UI will be hidden behind an `EmptyContent`: https://github.com/Automattic/wp-calypso/blob/8aae5dba70b51613d8cc918c4b1dc95179205f95/client/my-sites/domains/domain-search/domain-search.jsx#L126 But the TLDs are probably fetched in parallel and that causes this error. So it should be fine to just ignore this error, but please double-check that ;)